Lib/ufo2ft/featureWriters/markFeatureWriter.py

```diff
@@ -875,9 +876,39 @@ def _write(self):
             return False
 
         feaFile = self.context.feaFile
-        feaFile.statements.extend(newClassDefs)
-        # add empty line to separate classes from following statements
-        feaFile.statements.append(ast.Comment(""))
-        for _, feature in sorted(features.items()):
-            feaFile.statements.append(feature)
+
+        insertion_tag2index = ufo2ft.featureWriters.findFeatureInsertionMarkers(feaFile)
+
+        if insertion_tag2index:
+            # Write the class definitions at the first feature insertion point,
+            # no matter what it is.
```

Comment:
hm I forgot why we need to write class definitions "at the first feature insertion point, no matter what it is"?

This makes the finding-insertion-index logic global as opposed to independent for each writer (I would prefer the latter if possible).

Reply:
I think because there's no guarantee they're written anywhere before that 🤔